src/Command/LockCommand.php

```diff
@@ -37,64 +53,121 @@ protected function configure(): void
         }
         $contentType = $this->contentTypeRepository->findOneBy(['name' => $contentTypeName]);
         if (!$contentType instanceof ContentType) {
             throw new \RuntimeException('Content type not found');
         }
-        if (($time = \strtotime($timeArgument)) === false) {
-            throw new \RuntimeException('invalid time');
-        }
-        $by = $input->getOption('user');
+        $this->contentType = $contentType;
+
+        $by = $input->getOption(self::OPTION_USER);
         if (!\is_string($by)) {
             throw new \RuntimeException('Unexpected user name');
         }
-        $force = true === $input->getOption('force');
+        $this->by = $by;
 
-        $until = new \DateTime();
-        $until->setTimestamp($time);
+        if (null !== $input->getOption(self::OPTION_QUERY)) {
+            $rawQuery = \strval($input->getOption('query'));
+            $this->query = \json_decode($rawQuery, true);
+            if (\json_last_error() > 0) {
+                throw new \RuntimeException(\sprintf('Invalid json query %s', $rawQuery));
+            }
+        }
 
-        $io = new SymfonyStyle($input, $output);
+        $this->force = true === $input->getOption(self::OPTION_FORCE);
+    }
 
-        if ($input->getOption('if-empty') &&
-            0 !== $this->revisionRepository->findAllLockedRevisions($contentType, $by)->count()) {
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        if ($input->getOption(self::OPTION_IF_EMPTY) &&
+            0 !== $this->revisionRepository->findAllLockedRevisions($this->contentType, $this->by)->count()) {
             return 0;
         }
 
-        $ouuid = $input->getOption('ouuid') ? \strval($input->getOption('ouuid')) : null;
-        $rows = $this->revisionRepository->lockRevisions($contentType, $until, $by, $force, $ouuid);
+        if (null !== $input->getOption(self::OPTION_QUERY)) {
+            $search = $this->elasticaService->convertElasticsearchSearch([
+                'index' => (null !== $this->contentType->getEnvironment()) ? $this->contentType->getEnvironment()->getAlias() : '',
+                '_source' => false,
+                'body' => $this->query,
+            ]);
+
+            $documentCount = $this->elasticaService->count($search);
+            if (0 === $documentCount) {
+                $this->io->error(\sprintf('No documents found in %s with this query : %s', $this->contentType->getName(), \json_encode($this->query)));
 
-        if (0 === $rows) {
-            $io->error('no revisions locked, try force?');
+                return -1;
+            }
+
+            $revisionCount = 0;
+            foreach ($this->searchDocuments($search) as $document) {
+                $this->io->comment($document->getId());
+                $revisionCount += $this->revisionRepository->lockRevisions($this->contentType, $this->until, $this->by, $this->force, $document->getId());
```

Comment:
Maybe group the lock by 1000 ouuids ?

Reply:
Seems a good idea but I'm not really sure how to do that properly....

Reply:
In the searchDocuments function yields an array of documents at each scroll instead of one document at the time.